### PR TITLE
Revamp the doc

### LINKED ISSUE #23: Documentation
Is there any reason why the doc is in `.rst`? It doesn't render well on GitHub :/

### PR DESCRIPTION
**Pull request by theofidry**
Closes #23.

Main changes:

- Add a ToC
- Add links between each page of the doc to go the next or previous chapter
- Fix some RST things not supported by GitHub

### CODE REVIEW

File: resources/docs/usage.rst [hunk 1]
```diff
@@ -28,7 +28,7 @@ You can use the `Query Builder`_:
 
 .. caution::
 
-    Don't forget to `enable the DB aliases`_ if you want to use the ``DB``
+    Don't forget to [enable the DB aliases](configuration.rst) if you want to use the ``DB``
```
Comment (wouterj) [inline resources/docs/usage.rst]: you can't use markdown in rst. Does this work: `` `enable the DB aliases <configuration.rst>`_``?

File: resources/docs/migrations.rst [hunk 2]
```diff
@@ -129,5 +129,8 @@ Option                Description
 ``--seeder``          The class name of the seeder.
 ====================  =======================================================================
 
+Next chapter: [Usage](usage.rst)<br />
```
Comment (wouterj) [inline resources/docs/migrations.rst]: These links don't work either and you can't use HTML in rst. Let's use something like:

```rst
« `Usage <usage.rst>`_ | `Configuration <configuration.rst>`_ »
```

File: resources/docs/usage.rst (hunk 1 above)
Comment (theofidry) [inline resources/docs/usage.rst]: ah indeed, will try. This is also certainly why https://github.com/wouterj/WouterJEloquentBundle/pull/25#discussion_r81459631 fails

**Comment by theofidry** [comment]
@wouterj should be ok now

**Comment by wouterj** [comment]
Thanks!

**Pull request merged**